s3sync.go

```diff
@@ -512,6 +591,14 @@ func filterFilesForSync(sourceFileChan, destFileChan chan *fileInfo, del bool) c
 				}
 			}
 		}
+		if softDelete {
```

Comment:
This could be an else if for the previous statement. We likely dont want to delete and then try to soft delete.